pyfstat/make_sfts.py

```diff
@@ -92,31 +93,47 @@ def __init__(
         self.tbounds = [self.tstart, self.tend]
         logging.info("Using segment boundaries {}".format(self.tbounds))
 
-    def basic_setup(self):
+    def _get_sft_constraints_from_tstart_duration(self):
+        """
+        Use detectors, tstart and duration to set up a lalpulsar.SFTConstraints
+        object. This method is only used if noiseSFTs is not None.
+        """
+        SFTConstraint = lalpulsar.SFTConstraints()
+        SFTConstraint.detector = self.detectors
+
+        if self.tstart is None:
```

Comment:
double-checked: [XLALCWGPSinRange](https://lscsoft.docs.ligo.org/lalsuite/lalpulsar/group___s_f_tfile_i_o__h.html#gaca1afa637e9523e43ac631271cc34418) allows for either `minStartTime` or `maxStartTime` to be Null, so having only tstart but not `duration` here (the `elif` case) is indeed legal. Only setting `maxStartTime` would be possible too, but doesn't make sense with our `(tstart,duration)` input tuple.
So this looks allright!